src/test/java/unit/ReporterTest.java

```diff
@@ -41,6 +42,12 @@ public void deleteFile() {
         directory.delete();
     }
 
+    @Test
+    public void constructorNullTest() throws InvalidBrowserException {
+        // just verify no errors are thrown
+        new Reporter(null, null, null, null, null, null, null, null, null);
```

Comment:
what's the point of allowing Reporter to be instantiated with all null values?

Reply:
look at the changes to reporter. we don't want it to be able to, but it CAN, and I want to ensure it can still move forward. basically, this is to prevent dumb people from doing dumb things. May as well make it simpler for them